tools/test/audio/sof_test_perf_config.m

```diff
@@ -6,26 +6,40 @@
 %%
 
 %% Set user@domain for ssh, need to be set for playback test
-play.user = 'user@host.domain';
+% play.user = 'ubuntu@jf-apl-up2-pcm512x-1.jf.intel.com';
+play.user = 'test@10.223.163.42'; % UP2
+% play.user = 'root@10.223.165.220';  % RVP
```

Comment:
Can we pass the remote machine in on the command line and use the local machine if not provided by the user.

Reply:
Do you mean the user can use the local machine while passing the remote machine command line credentials [ aplay & arecord ] for playback and capture ?
Yes, the current setup is as shown above.

At the moment, I've set `play.ssh` and `rec.ssh` to 1, which means that a remote test machine can be used for quality testing on a local machine.

As a side note, use ssh-copy-id -f username@ip. to copy the files `To` &` from` remote machine to a local machine

Reply:
can matlab read environment variables from system? if yes, we can also use a environment variable to pass user and hostname to the script.

Reply:
Sure !! Thanks, I can create a new PR to integrate an environment variable from a command line parameter.

Reply:
Better to keep "user@host.domain" to not harass any real IP address and error when encountered. "localhost" would be safest but could result to unwanted playback on PC where run. I don't know which one is better.

Reply:
`test@something.local`

Reply:
just `play.user = 'localhost'` would do ssh to user's own account on same computer. It would work and not cause external traffic. But add comment into the same line to suggest `% username@ip-address`?

Reply:
> just `play.user = 'localhost'` would do ssh to user's own account on same computer. It would work and not cause external traffic. But add comment into the same line to suggest `% username@ip-address` syntax for ssh?

Done.

Reply:
> How often will people actually want to use `localhost`? If "very rarely" then it would be much better for the script to fail with an "unknown host something.local" error message (and ideally: a stack trace). This will point people directly at where they should make the change.
> 
> EDIT: of course a config file should be used eventually to avoid making `git status` "dirty" but not urgent.

Thanks @marc-hb , `Chao` suggested incorporating the audio quality script into daily CI automation.
I will build a new PR to enable automated support, keeping the above proposals and suggestions in mind. 
In the moment, emphasis is on addressing scripting issues in order to get the test functioning.

Reply:
Let me rephrase: what are the chances of someone actually wanting `localhost`?

Reply:
_what are the chances of someone actually wanting localhost_
`None.` In this situation, test execution is halted due to the notice "Waiting to connect to server" 

When you call an IP address on your computer, you are attempting to contact a target computer on the internet; but, when you call the IP address 127.0.0.1, you are interacting with a `local host`.
If we are unable to communicate with the target system, we will be unable to run the remote `test = remote_test_run(test), remote_copy_playback(), remote_capture(),remote_play(), remote_copy_capture()`

Reply:
> > what are the chances of someone actually wanting localhost
> None

Then don't use `localhost` as a default.

> When using localhost test execution is interrupted owing to the message "Waiting to connect to server". Therefore, the objective is to make sure that the user replaces localhost with the proper credentials of the target device.

"Waiting" sounds like there is a delay and the rest of the error message is not helpful either. Also, for some users `ssh localhost` will work and for other users it will not. So different users will have different symptoms in the same situation, not good from a "support" perspective. 

Maybe I miss something but I expect `ssh insert_DUT_here` to fail immediately and consistently with a useful message like "host insert_DUT_here not found". Then users can `git grep insert_DUT_here` and instantly land on the comments explaining what to do.

Did you try these and compare?

Reply:
Thanks !!

If you read the comment from Liam , "[Can we pass the remote machine in on the command line and use the local machine if not provided by the user.](https://github.com/thesofproject/sof/pull/6056#discussion_r934346836)" and Chao proposal is  to implement in CI daily automation,  

In my next PR, I'll include a matlab parser that reads command line argument parameters.  and set play.user = `remotemachineIP` instead of  `localhost`, so that out of box , user can run the test. 

In the moment, it is envisaged that the user will have to manually configure the external IP.